runtime/compiler/trj9/control/CompilationThread.cpp

```diff
@@ -10305,23 +10328,24 @@ TR::CompilationInfo::computeFreePhysicalMemory(bool &incompleteInfo)
 
             if (!incomplete)
                {
-               if (cached == 0 || availSwap == 0)
+               if (memInfo.cached == OMRPORT_MEMINFO_NOT_AVAILABLE)
```

Comment:
To minimize the number of `if` statements which are more expensive than other operations I am proposing the following code:
```
             uint64_t freePhysicalMemorySizeB = memInfo.availPhysical;
            
             if (memInfo.cached == OMRPORT_MEMINFO_NOT_AVAILABLE)
                incomplete = true;
             else
                freePhysicalMemorySizeB += memInfo.cached;
#if defined(LINUX)
             if (memInfo.buffered == OMRPORT_MEMINFO_NOT_AVAILABLE)
                incomplete = true;
             else
                freePhysicalMemorySizeB += memInfo.buffered;
#endif 
```